lib/vagrant-service-manager/service.rb

```diff
@@ -0,0 +1,34 @@
+# Loads all actions
+Dir["#{File.dirname(__FILE__)}/services/*.rb"].each { |f| require_relative f }
+
+module Vagrant
+  module ServiceManager
+    SUPPORTED_BOXES = ['adb', 'cdk']
+
+    class Service
+      def initialize(app, env)
+        @app = app
+        @machine = env[:machine]
+        @ui = env[:ui]
+        @services = @machine.config.servicemanager.services.split(',').map(&:chomp)
+        @docker_hook = Docker.new(@machine, @ui)
+        @openshift_hook = OpenShift.new(@machine, @ui)
+      end
+
+      def call(env)
+        if SUPPORTED_BOXES.include? @machine.guest.capability(:os_variant)
+          @app.call(env)
```

Comment:
This should be the first line in this method.

If our box is not supported then, it will break the middleware chain.